src/Common/src/System/Security/Cryptography/CngPkcs8.cs

```diff
@@ -250,8 +250,8 @@ internal static unsafe Pkcs8Response ImportPkcs8PrivateKey(ReadOnlySpan<byte> so
                     out rentWritten))
                 {
                     int size = rented.Length;
-                    ArrayPool<byte>.Shared.Return(rented);
-                    rented = ArrayPool<byte>.Shared.Rent(checked(size * 2));
+                    CryptoPool.Return(rented, rentWritten);
+                    rented = CryptoPool.Rent(checked(size * 2));
```

Comment:
If Rent fails (e.g. OOM), we could end up returning `rented` twice, I think, once here and once in the finally.